ansible/roles/artifacts/tasks/main.yaml

```
@@ -269,6 +269,42 @@
     mode: "644"
     checksum: sha256:3d93ca05f31b63424d7d7246a01a2365953705a0ed3323ba5b6fddd744a4bfea
 
+# tensorrt_rtmdet
+- name: Create tensorrt_rtmdet directory inside {{ data_dir }}
+  ansible.builtin.file:
+    path: "{{ data_dir }}/tensorrt_rtmdet"
+    mode: "755"
+    state: directory
+
+- name: Download tensorrt_rtmdet_onnx_models.tar.gz
+  become: true
+  ansible.builtin.get_url:
+    url: https://autoware-files.s3.us-west-2.amazonaws.com/models/tensorrt_rtmdet_onnx_models.tar.gz
+    dest: "{{ data_dir }}/tensorrt_rtmdet_onnx_models.tar.gz"
+    mode: "644"
+    checksum: sha256:eaf6fe9caf1b4a0211dc3b0f4068b1fd6c1d35fe1582cc2aed95b8ed9468c598
+
+- name: Extract tensorrt_rtmdet_onnx_models.tar.gz
+  ansible.builtin.unarchive:
+    src: "{{ data_dir }}/tensorrt_rtmdet_onnx_models.tar.gz"
+    dest: "{{ data_dir }}/tensorrt_rtmdet/"
+    remote_src: true
+
+- name: Move extracted files to the final destination
+  ansible.builtin.copy:
+    src: "{{ data_dir }}/tensorrt_rtmdet/tensorrt_rtmdet_onnx_models/"
+    dest: "{{ data_dir }}/tensorrt_rtmdet/"
+    mode: "0644"
+    remote_src: true
```

```suggestion
- name: Move extracted files to the final destination
  ansible.builtin.command: >
    mv {{ data_dir }}/tensorrt_rtmdet/tensorrt_rtmdet_onnx_models/* {{ data_dir }}/tensorrt_rtmdet/
  args:
    removes: "{{ data_dir }}/tensorrt_rtmdet/tensorrt_rtmdet_onnx_models/"
```

This way, we won't need the next step since it will be moved.

Also, `removes` argument tells Ansible to only execute the command if the specified path (in this case, `{{ data_dir }}/tensorrt_rtmdet/tensorrt_rtmdet_onnx_models/`) exists.

If the path does not exist, the task will be skipped, ensuring idempotency (i.e., running the playbook multiple times will not re-run this command unnecessarily).

---

The * wildcard doesn’t work with ansible.builtin.command, so I use ansible.builtin.shell. Is it okay for you?